src/creator.rs

```diff
@@ -97,17 +99,21 @@ impl<H: Hasher> Creator<H> {
 
         let prev_round_index = match round.checked_sub(1) {
             Some(prev_round) => prev_round as usize,
-            None => return,
+            None => {
+                Delay::new((self.create_lag)(round.into())).await;
+                return
+            },
         };
 
-        let delay = time::sleep((self.create_lag)(round.into()));
-        tokio::pin!(delay);
+        let mut delay = Delay::new((self.create_lag)(round.into())).fuse();
         loop {
-            if usize::from(round) < self.n_candidates_by_round.len() - 1 {
-                return; // Since we get unit from round r, we have enough units from previous rounds to create our unit 
+            if ((round + 2) as usize) < self.n_candidates_by_round.len() { // We need to require a number higher by one then currently highest round 
+                                                                           // (by 2 then length) to prevent attack when a malcious node is creating
+                                                                           // units without any delay to achive max_round as soon as possible
```

Comment:
Comments-to-the-right is an interesting style, but I don't really like the effect. This might be a personal preference though, so uh, @michalseweryn, @DamianStraszak what do you think?

Usually I support uniform style decisions, but comments' uniformity doesn't seem to be crucial, so I would also be fine if we settled on "any comment style is fine".

Reply:
Ugh, please uniform comment style... I like the effect, but since throughout the repos we use different style, I don't think we should incorporate it. 
I am fine with the default way, where a comment is above the commented entity.

Reply:
Yeah, I would prefer comment above the if -- we do it consistently this way.